packages/core/src/instance-settings/instance-settings.ts

```diff
@@ -225,6 +228,14 @@ export class InstanceSettings {
 			.digest('hex');
 	}
 
+	private getOrGenerateHmacSignatureSecret() {
+		const hmacSignatureSecretFromEnv = process.env.N8N_HMAC_SIGNATURE_SECRET;
```

Comment:
just for my understanding, why would using the encryptionKey (or a hash of it) be considered insecure? Introducing new env vars adds configuration complexity and I think we should avoid it unless really necessary.